src/map/status.c

```diff
@@ -10507,273 +10507,319 @@ static void status_change_start_stop_action(struct block_list *bl, enum sc_type
+		status_change_end(bl, SC_ATTHASTE_INFINITY, INVALID_TIMER);
+		break;
+	// Group A Status (doesn't overlap)
+	case SC_SWING:
+	case SC_SYMPHONY_LOVE:
+	case SC_MOONLIT_SERENADE:
+	case SC_RUSH_WINDMILL:
+	case SC_ECHOSONG:
+	case SC_HARMONIZE:
+		if (type != SC_SWING)
+			status_change_end(bl, SC_SWING, INVALID_TIMER);
+
+		if (type != SC_SYMPHONY_LOVE)
+			status_change_end(bl, SC_SYMPHONY_LOVE, INVALID_TIMER);
+
+		if (type != SC_MOONLIT_SERENADE)
+			status_change_end(bl, SC_MOONLIT_SERENADE, INVALID_TIMER);
+
+		if (type != SC_RUSH_WINDMILL)
+			status_change_end(bl, SC_RUSH_WINDMILL, INVALID_TIMER);
+
+		if (type != SC_ECHOSONG)
+			status_change_end(bl, SC_ECHOSONG, INVALID_TIMER);
+
+		if (type != SC_HARMONIZE)
+			status_change_end(bl, SC_HARMONIZE, INVALID_TIMER);
+
+		break;
+	// Group B Status
+	case SC_SIREN:
+	case SC_DEEP_SLEEP:
+	case SC_SIRCLEOFNATURE:
+	case SC_LERADS_DEW:
+	case SC_MELODYOFSINK:
+	case SC_BEYOND_OF_WARCRY:
+	case SC_UNLIMITED_HUMMING_VOICE:
+	case SC_GLOOMYDAY:
+	case SC_SONG_OF_MANA:
+	case SC_DANCE_WITH_WUG:
+		if (type != SC_SIREN)
+			status_change_end(bl, SC_SIREN, INVALID_TIMER);
+
+		if (type != SC_DEEP_SLEEP)
+			status_change_end(bl, SC_DEEP_SLEEP, INVALID_TIMER);
+
+		if (type != SC_SIRCLEOFNATURE)
+			status_change_end(bl, SC_SIRCLEOFNATURE, INVALID_TIMER);
+
+		if (type != SC_LERADS_DEW)
+			status_change_end(bl, SC_LERADS_DEW, INVALID_TIMER);
+
+		if (type != SC_MELODYOFSINK)
+			status_change_end(bl, SC_MELODYOFSINK, INVALID_TIMER);
+
+		if (type != SC_BEYOND_OF_WARCRY)
+			status_change_end(bl, SC_BEYOND_OF_WARCRY, INVALID_TIMER);
+
+		if (type != SC_UNLIMITED_HUMMING_VOICE)
+			status_change_end(bl, SC_UNLIMITED_HUMMING_VOICE, INVALID_TIMER);
+
+		if (type != SC_GLOOMYDAY)
+			status_change_end(bl, SC_GLOOMYDAY, INVALID_TIMER);
+
+		if (type != SC_SONG_OF_MANA)
+			status_change_end(bl, SC_SONG_OF_MANA, INVALID_TIMER);
+
+		if (type != SC_DANCE_WITH_WUG)
+			status_change_end(bl, SC_DANCE_WITH_WUG, INVALID_TIMER);
+
+		break;
+	case SC_REFLECTSHIELD:
+		status_change_end(bl, SC_LG_REFLECTDAMAGE, INVALID_TIMER);
+		break;
+	case SC_LG_REFLECTDAMAGE:
+		status_change_end(bl, SC_REFLECTSHIELD, INVALID_TIMER);
+		break;
+	case SC_SHIELDSPELL_DEF:
+	case SC_SHIELDSPELL_MDEF:
+	case SC_SHIELDSPELL_REF:
+		status_change_end(bl, SC_MAGNIFICAT, INVALID_TIMER);
+		status_change_end(bl, SC_SHIELDSPELL_DEF, INVALID_TIMER);
+		status_change_end(bl, SC_SHIELDSPELL_MDEF, INVALID_TIMER);
+		status_change_end(bl, SC_SHIELDSPELL_REF, INVALID_TIMER);
+		break;
+	case SC_GENTLETOUCH_ENERGYGAIN:
+	case SC_GENTLETOUCH_CHANGE:
+	case SC_GENTLETOUCH_REVITALIZE:
+		if (type != SC_GENTLETOUCH_REVITALIZE)
+			status_change_end(bl, SC_GENTLETOUCH_REVITALIZE, INVALID_TIMER);
+
+		if (type != SC_GENTLETOUCH_ENERGYGAIN)
+			status_change_end(bl, SC_GENTLETOUCH_ENERGYGAIN, INVALID_TIMER);
+
+		if (type != SC_GENTLETOUCH_CHANGE)
+			status_change_end(bl, SC_GENTLETOUCH_CHANGE, INVALID_TIMER);
+
+		break;
```

Comment:
why there are so many empty lines ?
I guess its still editing

Reply:
Nope, that's final. Those blank lines improve the readability. (At least in my opinion.)

Reply:
I recheck the hercules coding style and ...
it only says leave empty lines for comments ... and each every new function

these line break ... doesn't fall into either of these ...
well if the code doesn't break then I trust you =/

Reply:
The code wont break. It's just a clear visual separation between the `if`-clause and leaving the `switch`-case.